snapshots/devmapper/pool_device.go

```diff
@@ -306,6 +306,20 @@ func (p *PoolDevice) CreateSnapshotDevice(ctx context.Context, deviceName string
 		return metaErr
 	}
 
+	if p.IsLoaded(deviceName) {
+		log.G(ctx).Debugf("suspending %q", deviceName)
+		suspendErr := p.SuspendDevice(ctx, deviceName)
+		if suspendErr != nil {
+			return suspendErr
+		}
+		defer func() {
+			err := p.ResumeDevice(ctx, deviceName)
+			if err != nil {
+				log.G(ctx).WithError(err).Errorf("failed to resume %q", baseInfo.Name)
```

Comment:
`p.ResumeDevice` already returns `failed to resume %q` error message, so there no need to dup it. 
Instead, can we be more descriptive here? Like `failed to resume base device %q after taking snapshot %q` with fields: base name / id, snap name / id. This should ease troubleshooting.
(same applies to suspend)

Also could you please add a comment and the "according to link` mentioned in the PR description to explain why're we doing this here.

Reply:
I've included the link and elaborate on the log message. Regarding SuspendDevice, since the function is returning the error (unlike ResumeDevice in defer), I don't think we need to log the error inside the function. What do you think?

Reply:
works for me